### PR TITLE
Change to Travis Script and write debug logs only if we build for Debug config

### PR DESCRIPTION
**Pull request by kchitalia**

### CODE REVIEW

**Comment by kchitalia** [comment]
@huminzhi I am checking in the script for travis. We can look at filtering the logs further.

File: .travis.yml [hunk 1]
```diff
@@ -4,14 +4,14 @@ branches:
      - unstable
 
 language: objective-c
-osx_image: xcode7
+osx_image: xcode7.2
```
Comment (bhariharan) [inline .travis.yml]: We should probably use the latest (```7.3```).

File: .travis.yml [hunk 2]
```diff
@@ -4,14 +4,14 @@ branches:
...
 
 env:
   matrix:
-    - SDK=iphonesimulator OS=8.4 NAME='iphone 6'
-    - SDK=iphonesimulator9.1 OS=9.1 NAME='iphone 6'
-    - SDK=iphonesimulator OS=8.4 NAME='ipad 2'
-    - SDK=iphonesimulator9.1 OS=9.1 NAME='ipad 2'
+    - SDK=iphonesimulator OS=8.4 NAME="iPhone 6"
+    - SDK=iphonesimulator9.2 OS=9.2 NAME="iPhone 6"
```
Comment (bhariharan) [inline .travis.yml]: Should probably use the latest here too (```9.3```).

File: .travis.yml (hunk 1 above)
Comment (kchitalia) [inline .travis.yml]: There is a bug in travis CI that causes the build and tests to freeze if xcode version is 7.3 and iOS version is 9.3. Till the bug is fixed we'll have to downgrade to 7.2 and iOS 9.2

File: .travis.yml (hunk 2 above)
Comment (kchitalia) [inline .travis.yml]: Same here.

File: libs/SalesforceRestAPI/SalesforceRestAPITests/SalesforceRestAPITests.m [hunk 3]
```diff
@@ -45,7 +45,11 @@ @implementation SalesforceRestAPITests
 + (void)setUp
 {
     @try {
-        [SFLogger setLogLevel:SFLogLevelDebug];
+        #if defined(DEBUG)
```
Comment (bhariharan) [inline libs/SalesforceRestAPI/SalesforceRestAPITests/SalesforceRestAPITests.m]: Don't we always want log output from tests?

Comment (kchitalia) [inline libs/SalesforceRestAPI/SalesforceRestAPITests/SalesforceRestAPITests.m]: Not if we are running them is release mode, right?

Comment (bhariharan) [inline libs/SalesforceRestAPI/SalesforceRestAPITests/SalesforceRestAPITests.m]: Release binaries are never built for the test apps or test targets AFAIK. I don't think we need these conditionals in the test classes.

**Comment by kchitalia** [comment]
@bhariharan Removed the #if def from test classes.

**Comment by bhariharan** [comment]
👍

**Pull request merged**